myjdapi/myjdapi.py

```diff
@@ -786,74 +819,81 @@ def get_device(self, device_name=None, device_id=None):
+            if params is not None:
+                for param in params:
+                    if not isinstance(param, list):
+                        # params_request+=[str(param).replace("'",'\"').replace("True","true").replace("False","false").replace('None',"null")]
+                        params_request += [json.dumps(param)]
+                    else:
+                        params_request += [param]
+            params_request = {"apiVer": self.__api_version, "url": path, "params": params_request,
+                              "rid": self.__request_id}
+            data = json.dumps(params_request).replace('"null"', "null").replace("'null'", "null")
+            encrypted_data = self.__encrypt(self.__device_encryption_token, data)
             if action is not None:
-                request_url=self.__api_url+action+path
+                request_url = api + action + path
             else:
-                request_url=self.__api_url+path
-            encrypted_response = requests.post(request_url,headers={"Content-Type": "application/aesjson-jd; charset=utf-8"},data=encrypted_data)
+                request_url = api + path
+            try:
+                encrypted_response = requests.post(request_url,
+                                                   headers={"Content-Type": "application/aesjson-jd; charset=utf-8"},
+                                                   data=encrypted_data)
+            except requests.exceptions.RequestException as e:
+                return None
         if encrypted_response.status_code != 200:
-            error_msg=json.loads(encrypted_response.text)
-            msg="\n\tSOURCE: "+error_msg["src"]+"\n\tTYPE: "+ \
-                                error_msg["type"]+"\n------\nREQUEST_URL: "+ \
-                                self.__api_url+path
+            error_msg = json.loads(encrypted_response.text)
+            msg = "\n\tSOURCE: " + error_msg["src"] + "\n\tTYPE: " + \
+                  error_msg["type"] + "\n------\nREQUEST_URL: " + \
+                  api + path
             if http_method == "GET":
-                msg+=query
-            msg+="\n"
+                msg += query
+            msg += "\n"
             if data is not None:
-                msg+="DATA:\n"+data
-            raise(MYJDException(msg))
+                msg += "DATA:\n" + data
+            raise (MYJDException(msg))
         if action is None:
             if not self.__server_encryption_token:
                 response = self.__decrypt(self.__login_secret, encrypted_response.text)
             else:
                 response = self.__decrypt(self.__server_encryption_token, encrypted_response.text)
         else:
-            if params is not None:
-                response = self.__decrypt(self.__device_encryption_token, encrypted_response.text)
-            else:
-                return {"data" : response}
+            response = self.__decrypt(self.__device_encryption_token, encrypted_response.text)
```

Comment:
I have to check this. I can't remember exactly what it required this but I think there was some api response that wasn't encrypted maybe it's not the case anymore. (A comment by me about why I did this, at the time, would have been nice :P but I didn't do it)

Reply:
Ye, the {"data" : response} was invalid in my case it didn't exist and it needed to get the response ^^